haystack/nodes/document_classifier/base.py

```diff
@@ -17,35 +17,25 @@ class BaseDocumentClassifier(BaseComponent):
     query_count = 0
     query_time = 0
 
-    def __init__(
-        self,
-        classification_field: str = None,
-        convert_to_dicts: bool = False
-    ):
-        self.classification_field = classification_field
-        self.convert_to_dicts = convert_to_dicts
-
     @abstractmethod
     def predict(self, documents: List[Document]):
         pass
 
-    def run(self, documents: Union[List[dict], List[Document]]): # type: ignore
+    def run(self, documents: Union[List[dict], List[Document]], root_node: str): # type: ignore
         self.query_count += 1
-        field_map = {}
-        if self.classification_field is not None:
-            field_map = {self.classification_field: "content"}
         if documents:
             predict = self.timing(self.predict, "query_time")            
-            documents = [Document.from_dict(d, field_map=field_map) if isinstance(d, dict) else d for d in documents]
+            documents = [Document.from_dict(doc) if isinstance(doc, dict) else doc for doc in documents]
             results = predict(documents=documents)
         else:
             results = []
 
         document_ids = [doc.id for doc in results]
         logger.debug(f"Retrieved documents with IDs: {document_ids}")
 
-        if self.convert_to_dicts:
-            results = [d.to_dict(field_map=field_map) for d in results]
+        # convert back to dicts if we are in an indexing pipeline
+        if root_node == "File":
+            results = [doc.to_dict() for doc in results]
```

Comment:
At some point in the future we should probably also switch over the indexing pipelines to work with Documents rather than dicts (or be compliant to both). Not in the scope of this PR though, so the conversion approach is totally right here

Reply:
Yes that would make a lot much easier. However we might want to investigate first if we lose some flexibility somewhere. I could image that self-written preprocessing nodes might need to be changed as well, but this should not be too effortful.